NExtends/Primitives/Strings/String.extensions.cs

```diff
@@ -0,0 +1,564 @@
+                if (rest != 0)
+                {
+                    return false;
+                }
+            }
+            else if (!string.IsNullOrEmpty(match.Groups["temp"].Value))
+            {
+                // Si le numéro est de type temporaire, on impose la clé (donc ça doit passer dans le if au-dessus)
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// On passe par un MailAddress, plus simple
+        /// </summary>
+        public static bool isEmail(this string email)
+        {
+            //https://stackoverflow.com/a/1374644/533686
+            try
+            {
+                var addr = new System.Net.Mail.MailAddress(email);
+                return addr.Address == email;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public static bool ContainsIgnoreCase(this string source, string value)
+        {
+            return source.IndexOf(value, StringComparison.OrdinalIgnoreCase) != -1;
+        }
+
+        public static String SQLProtect(this String value)
+        {
+            return value.Replace("'", "''");
+        }
+
+        public static String ToFirstUpper(this String word)
+        {
+            if (!String.IsNullOrEmpty(word))
+            {
+                return word[0].ToString().ToUpper() + word.Substring(1);
+            }
+            else
+            {
+                return word;
+            }
+        }
+
+        public static String JSONProtect(this String value)
+        {
+            if (value != null)
+            {
+                String json = value.Replace("\r\n", "<br>");
+                json = json.Replace("\n", "<br>");
+                json = json.Replace("\r", "<br>");
+
+                json = json.Replace("\\", "\\\\");
+
+                json = json.Replace("'", "\\'");
+                json = json.Replace("\"", "\\\"");
+
+                return json;
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        public static String JSONQuotesProtect(this String value)
+        {
+            if (value != null)
+            {
+                String json = value.Replace("\r\n", "<br>");
+                json = json.Replace("\n", "<br>");
+                json = json.Replace("\r", "<br>");
+
+                json = json.Replace("\\", "\\\\");
+
+                json = json.Replace("\"", "\\\"");
+
+                return json;
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        //Quelqueq méthodes pour convertir des types basiques en JSON
+        public static String ToJSON(this bool b)
+        {
+            return b.ToString().ToLower();
+        }
+        public static String ToJSON(this String s)
+        {
+            return JsonConvert.SerializeObject(s);//  "\"" + s.JSONQuotesProtect() + "\"";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)